mantidproject/mantid
# SortXAxis Vector Subscript out of range error

Only test on windows, and on Debug mode
------------------------------------------------
**Description of work.**
This PR fixes a vector subscript out of range error in the `SortXAxis` algorithm. This causes old TOSCA workflows to fail.
A system test is also included to try and provide coverage for older TOSCA run numbers.

**To test:**
1.`Interfaces`->`Indirect`->`Data Reduction`->`ISISEnergyTransfer` tab
2. Instrument is `TOSCA`
3. Enter run number `4970`
4. Click `Run` and wait
There should be no error.

Fixes #25396 and #25345 

*No release notes required as this affects older data and occurs only for windows users*

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
